components/app/myrw/widgets/pages/edit.js

```diff
@@ -419,57 +289,16 @@ WidgetsEdit.propTypes = {
   user: PropTypes.object.isRequired,
   locale: PropTypes.string.isRequired,
   // ACTIONS
-  setFilters: PropTypes.func.isRequired,
-  setSize: PropTypes.func.isRequired,
-  setColor: PropTypes.func.isRequired,
-  setCategory: PropTypes.func.isRequired,
-  setValue: PropTypes.func.isRequired,
-  setOrderBy: PropTypes.func.isRequired,
-  setAggregateFunction: PropTypes.func.isRequired,
-  setLimit: PropTypes.func.isRequired,
-  setChartType: PropTypes.func.isRequired,
-  setVisualizationType: PropTypes.func.isRequired,
-  setAreaIntersection: PropTypes.func.isRequired,
-  setBand: PropTypes.func.isRequired,
-  setLayer: PropTypes.func.isRequired,
-  setDataset: PropTypes.func.isRequired,
-  setTitle: PropTypes.func.isRequired,
-  setZoom: PropTypes.func.isRequired,
-  setLatLng: PropTypes.func.isRequired,
-  setEmbed: PropTypes.func.isRequired
+  setDataset: PropTypes.func.isRequired
 };
 
 const mapStateToProps = state => ({
   user: state.user,
-  widgetEditor: state.widgetEditor,
   locale: state.common.locale
 });
 
 const mapDispatchToProps = dispatch => ({
-  setFilters: filter => dispatch(setFilters(filter)),
-  setColor: color => dispatch(setColor(color)),
-  setSize: size => dispatch(setSize(size)),
-  setCategory: category => dispatch(setCategory(category)),
-  setValue: value => dispatch(setValue(value)),
-  setOrderBy: value => dispatch(setOrderBy(value)),
-  setAggregateFunction: value => dispatch(setAggregateFunction(value)),
-  setLimit: value => dispatch(setLimit(value)),
-  setChartType: value => dispatch(setChartType(value)),
-  setVisualizationType: vis => dispatch(setVisualizationType(vis)),
-  setAreaIntersection: value => dispatch(setAreaIntersection(value)),
-  setBand: band => dispatch(setBand(band)),
-  setTitle: title => dispatch(setTitle(title)),
-  setLayer: (layerId) => {
-    new LayersService()
-      .fetchData({ id: layerId })
-      .then(layer => dispatch(setLayer(layer)))
-      // TODO: better handling of the error
-      .catch(err => toastr.error('Error', err));
-  },
-  setDataset: dataset => dispatch(setDataset(dataset)),
-  setZoom: zoom => dispatch(setZoom(zoom)),
-  setLatLng: latLng => dispatch(setLatLng(latLng)),
-  setEmbed: embed => dispatch(setEmbed(embed))
+  setDataset: dataset => dispatch(setDataset(dataset))
```

Comment:
Don't use mapDispatchToProps this way. Let's use always this one, unless you need something special. We will avoid bugs because you missed to pass the correct prop or something like that

```
const mapDispatchToProps = {
  setDataset
}
```